[JIT] Add torch.classes.load_library

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #32555 [JIT] Serialize attributes and types in ClassType serialization
* #32553 [JIT] Support for registering C++ lambdas as methods on custom C++ class
* #32514 Revert "Remove __torch__ from custom class qualname"
* **#32508 [JIT] Add torch.classes.load_library**
* #32477 [JIT] Support returning tuple from custom bound C++ method
* #32471 [JIT] Fix custom class method binding for const methods
* #32470 [JIT] Test __getstate__ and __setstate__ for custom bound C++ classes
* #32260 [JIT] Passing custom class as arg

Differential Revision: [D19525175](https://our.internmc.facebook.com/intern/diff/D19525175)

Commit:
[JIT] Add torch.classes.load_library

[ghstack-poisoned]

dzhulgakov reviewed (commented):
It's weird that `torch.ops.load_library` and `torch.classes.load_library` do the same thing but separately. I think it's better to just keep only one. Or in the worst case you can just delegate to torch.ops directly - why copy paste?

kostmo commented:
## :pill: CircleCI build failures summary and remediations
As of commit f1aeda0c:


**None of the build failures appear to be your fault**.

* **1/1** [broken upstream](https://dr.pytorch.org/master-timeline.html?count=1&sha1=1c017f0c14c91bd5125ab387a90441b0c0e2f3ad&should_suppress_scheduled_builds=true&should_suppress_fully_successful_columns=true&max_columns_suppress_successful=35) at merge base 1c017f0c since Jan 22
  <details>
  <summary>
  Please rebase on the <code>viable/strict</code> branch (expand for instructions)
  </summary>
  
  If your commit is newer than `viable/strict`, you can try basing on an older, stable commit:
  ```
  git fetch origin viable/strict
  git rebase --onto viable/strict $(git merge-base origin/master HEAD)
  ```
  If your commit is older than `viable/strict`:
  ```
  git fetch origin viable/strict
  git rebase viable/strict
  ```
  Check out the [recency history](https://dr.pytorch.org/master-viable-commits.html) of this "viable master" tracking branch.
  
  </details>
  


## Detailed failure analysis

One may explore the probable reasons each build failed interactively [on the Dr. CI website](https://dr.pytorch.org/commit-details.html?sha1=f1aeda0cc7b00adc5511c617e40e919016d23244).

### :construction: 1 upstream failure recognized by patterns:
These builds matched patterns, but were probably **caused by upstream breakages**:
* [pytorch_macos_10_13_py3_test](https://dr.pytorch.org/build-details.html?build_id=89538250) from Jan 22 until Jan 22



---

<details><summary>This comment was automatically generated by <a href="https://github.com/kostmo/circleci-failure-tracker/tree/master/docs/from-pull-request-comment">Dr. CI</a> (expand for details).</summary>Follow <a href="https://dr.pytorch.org/admin/comments-opt-out.html">this link to opt-out</a> of these comments for your Pull Requests.

Please report bugs/suggestions on the <a href="https://github.com/kostmo/circleci-failure-tracker/issues">GitHub issue tracker</a>.
</details>

Commit:
Update on "[JIT] Add torch.classes.load_library"


Differential Revision: [D19525175](https://our.internmc.facebook.com/intern/diff/D19525175)

[ghstack-poisoned]

Commit:
Update on "[JIT] Add torch.classes.load_library"


Differential Revision: [D19525175](https://our.internmc.facebook.com/intern/diff/D19525175)

[ghstack-poisoned]

Commit:
Update on "[JIT] Add torch.classes.load_library"


Differential Revision: [D19525175](https://our.internmc.facebook.com/intern/diff/D19525175)

[ghstack-poisoned]

Commit:
Update on "[JIT] Add torch.classes.load_library"


Differential Revision: [D19525175](https://our.internmc.facebook.com/intern/diff/D19525175)

[ghstack-poisoned]

dzhulgakov reviewed (approved):
Ok, that works :)

Btw, maybe copy the docstring too?

Commit:
Update on "[JIT] Add torch.classes.load_library"


Differential Revision: [D19525175](https://our.internmc.facebook.com/intern/diff/D19525175)

[ghstack-poisoned]

Commit:
Update on "[JIT] Add torch.classes.load_library"


Differential Revision: [D19525175](https://our.internmc.facebook.com/intern/diff/D19525175)

[ghstack-poisoned]
